packages/client/src/views/SysAdmin/Config/DynamicModal.tsx

```diff
@@ -0,0 +1,242 @@
+const Field = styled.div`
+  margin-bottom: 30px;
+  @media (max-width: ${({ theme }) => theme.grid.breakpoints.lg}px) {
+    margin-bottom: 0px;
+  }
+`
+const HalfWidthInput = styled(TextInput)`
+  width: 300px;
+  @media (max-width: ${({ theme }) => theme.grid.breakpoints.md}px) {
+    width: 100%;
+  }
+`
+const ErrorMessage = styled.div`
+  position: relative;
+  ${({ theme }) => theme.fonts.semiBoldFont};
+  color: ${({ theme }) => theme.colors.error};
+  margin-left: 6px;
+`
+type IApplicationName = {
+  applicatioName: string
+}
+type State = {
+  applicationName: string
+  updatingValue: boolean
+  errorOccured: boolean
+  errorMessages: string
+}
+interface IProps {
+  changeModalName: string
+  showNotification: boolean
+  hideModal: () => void
+  valueChanged: (
+    notificationStatus: NOTIFICATION_TYPE,
+    messages: string
+  ) => void
+}
+
+type DispatchProps = {
+  updateConfig: () => void
+}
+
+type IFullProps = IProps & IntlShapeProps & DispatchProps
+
+function isApplicationName(
+  modalName: string,
+  value: IApplicationName
+): value is IApplicationName {
+  return modalName === GeneralActionId.APPLICATION_NAME
+}
+class DynamicModalComponent extends React.Component<IFullProps, State> {
+  constructor(props: IFullProps) {
+    super(props)
+    this.state = {
+      applicationName: EMPTY_STRING,
+      updatingValue: false,
+      errorOccured: false,
+      errorMessages: EMPTY_STRING
+    }
+  }
+
+  showChangeModal: boolean = !!!this.props.changeModalName ? false : true
+
+  setApplicationName = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value
+    this.setState(() => ({
+      applicationName: value
+    }))
+  }
+
+  mutationHandler(
+    modalName: string,
+    value: IApplicationName,
+    valueChanged: (
+      notificationStatus: NOTIFICATION_TYPE,
+      messages: string
+    ) => void
+  ) {
+    if (isApplicationName(modalName, value)) {
+      this.callUpdateApplicationNameMutation(value.applicatioName)
+        .then(() => {
+          valueChanged(
+            NOTIFICATION_TYPE.SUCCESS,
+            this.props.intl.formatMessage(
+              messages.applicationNameChangeNotification
+            )
+          )
+        })
+        .catch(() => {
+          this.setState({
+            errorOccured: true,
+            errorMessages: this.props.intl.formatMessage(
+              messages.applicationNameChangeError
+            )
+          })
+          valueChanged(
+            NOTIFICATION_TYPE.ERROR,
+            this.props.intl.formatMessage(messages.applicationNameChangeError)
+          )
+        })
+    }
+  }
+
+  async callUpdateApplicationNameMutation(applicationName: string) {
+    try {
+      this.setState({ updatingValue: true })
+      const res = await configApplicationMutations.updateApplicationName(
+        applicationName
+      )
+      if (res && res.data) {
+        this.setState({ updatingValue: false })
+        this.props.updateConfig()
```

Comment:
We are already getting the updated `ApplicationConfig` in `res.data` after calling the mutation. I think we can pass this data to `updateConfig` and save that in both Redux and IndexedDB.

Reply:
Fixed. Thanks for addressing this issue.